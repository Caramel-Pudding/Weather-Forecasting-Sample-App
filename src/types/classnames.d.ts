declare module "classnames" {
  function classNames(...classes: string[]): string;
  export default classNames;
}
