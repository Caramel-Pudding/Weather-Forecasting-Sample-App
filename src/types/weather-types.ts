export interface City {
  id: number;
  country: string;
  name: string;
}

export interface WeatherData {
  city: City;
}
