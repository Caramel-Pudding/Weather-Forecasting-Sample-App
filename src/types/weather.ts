export enum PartOfTheDay {
  n = "night",
  d = "day",
}

export enum WeatherType {
  Clear = "Clear",
  Rain = "Rain",
  Snow = "Snow",
  Extreme = "Extreme",
}

export interface City {
  id: number;
  country: string;
  name: string;
}

export interface WeatherListItem {
  clouds: {
    all: number;
  };
  dt: number;
  dt_txt: string;
  main: {
    grnd_level: number;
    humidity: number;
    pressure: number;
    sea_level: number;
    temp: number;
    temp_kf: number;
    temp_max: number;
    temp_min: number;
  };
  rain?: {
    "3h"?: number;
  };
  snow?: {
    "3h"?: number;
  };
  sys: {
    pod: PartOfTheDay;
  };
  weather: {
    description: string;
    icon: string;
    id: number;
    main: WeatherType;
  }[];
  wind: {
    deg: number;
    speed: number;
  };
}

export interface WeatherData {
  city: City;
  cnt: number;
  cod: string;
  message: number;
  list: WeatherListItem[];
}
