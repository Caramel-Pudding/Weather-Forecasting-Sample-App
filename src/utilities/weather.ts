import { WeatherListItem } from "@/types/weather";

export const getWeatherListForToday = (
  weatherList: WeatherListItem[]
): WeatherListItem[] => {
  if (!weatherList || weatherList.length === 0) {
    return [];
  }
  const todayDate = new Date(weatherList[0].dt_txt).getDate();
  return weatherList.filter(
    (listItem) => new Date(listItem.dt_txt).getDate() === todayDate
  );
};

export const getTimeStringFromWeatherTimestamp = (
  weatherTimestamp: string
): string => {
  const timestamp = new Date(weatherTimestamp);
  return `${timestamp.getHours()}:00`;
};
