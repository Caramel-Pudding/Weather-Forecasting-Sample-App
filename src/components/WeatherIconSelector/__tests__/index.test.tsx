import React from "react";
import { render, cleanup } from "@/tests/test-utils";

import { WeatherType } from "@/types/weather";

import { WeatherIconSelector } from "..";

describe("WeatherIconSelector Component", () => {
  afterEach(cleanup);

  it("Should render clear icon if weather is clear", () => {
    // * #TEST: ARRANGE
    const { container, getByAltText } = render(
      <WeatherIconSelector weatherType={WeatherType.Clear} />,
      {}
    );
    const icon = getByAltText("Clear");
    // * #TEST: ASSERT
    expect(icon).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("Should render cloudy icon if weather is rainy", () => {
    // * #TEST: ARRANGE
    const { container, getByAltText } = render(
      <WeatherIconSelector weatherType={WeatherType.Rain} />,
      {}
    );
    const icon = getByAltText("Cloudy");
    // * #TEST: ASSERT
    expect(icon).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
