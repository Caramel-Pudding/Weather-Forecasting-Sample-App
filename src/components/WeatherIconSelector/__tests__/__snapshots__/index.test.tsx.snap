// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherIconSelector Component Should render clear icon if weather is clear 1`] = `
<div>
  <img
    alt="Clear"
    src="icons/weather-sun.svg"
  />
</div>
`;

exports[`WeatherIconSelector Component Should render cloudy icon if weather is rainy 1`] = `
<div>
  <img
    alt="Cloudy"
    src="icons/weather-cloud.svg"
  />
</div>
`;
